src/test/java/com/pgs/booking/service/UserServiceTest.java

```diff
@@ -1,50 +1,164 @@
+import com.pgs.booking.model.entity.Role;
+import com.pgs.booking.model.entity.User;
+import com.pgs.booking.repository.RoleRepository;
 import com.pgs.booking.repository.UserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
-import org.mockito.InjectMocks;
 import org.mockito.Mock;
+import org.mockito.Spy;
 import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.security.core.Authentication;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.security.oauth2.provider.OAuth2Authentication;
 import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.oauth2.provider.token.store.InMemoryTokenStore;
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
 
+import java.util.List;
+import java.util.Optional;
+
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.*;
 
 @ExtendWith(MockitoExtension.class)
 class UserServiceTest {
 
-    private static final User USER = User.builder()
+    private static final Role ROLE_1 = Role.builder()
+            .name("USER")
+            .build();
+    private static final Role ROLE_2 = Role.builder()
+            .name("ADMIN")
+            .build();
+    private static final RoleDto ROLE_DTO = RoleDto.builder()
+            .id(3L)
+            .name("USER")
+            .build();
+    private static final List<Role> ROLE_LIST = List.of(ROLE_1);
+    private static final List<RoleDto> ROLE_DTO_LIST = List.of(ROLE_DTO);
+    private static final User USER_1 = User.builder()
+            .id(2L)
+            .username("user1")
+            .roles(ROLE_LIST)
+            .build();
+    private static final User USER_2 = User.builder()
+            .id(3L)
+            .username("user2")
+            .enabled(true)
+            .roles(ROLE_LIST)
+            .build();
+    private static final CreateUserDto CREATE_USER_DTO = CreateUserDto.builder()
             .username("user")
+            .roles(ROLE_DTO_LIST)
             .build();
     @Mock
     private UserRepository userRepository;
     @Mock
-    private TokenStore tokenStore;
-    @InjectMocks
+    private RoleRepository roleRepository;
     private UserService userService;
 
+    private RoleEntityMapper roleEntityMapper = new RoleEntityMapper();
+    private RoleDtoMapper roleDtoMapper = new RoleDtoMapper();
+    @Spy
+    private TokenStore tokenStore = new InMemoryTokenStore();
+    private CreateUserDtoMapper createUserDtoMapper = new CreateUserDtoMapper(roleEntityMapper);
+    private UserDtoMapper userDtoMapper = new UserDtoMapper(roleDtoMapper);
+
+    @BeforeEach
+    void setUp() {
+        userService = new UserService(userRepository,
```

Comment:
This can be done by `MockitoExtension` if you annotate `UserService` with `@InjectMocks` and `@Spy` to the added mappers fields.